src/array_api_stubs/_draft/creation_functions.py

```diff
@@ -645,3 +646,29 @@ def zeros_like(
     out: array
         an array having the same shape as ``x`` and filled with zeros.
     """
+
+
+def from_binsparse(arrays: dict[str, array], descriptor: dict, /) -> array:
+    """
+    Returns a new array containing the data from another (array) object with a ``__binsparse__`` method.
+
+    Parameters
+    ----------
+    arrays: dict[str, array]
+        input constituent arrays.
+    descriptor: dict
+        The parsed binsparse descriptor of the array.
+
+    Returns
+    -------
+    out: array
+        an array containing the data in `arrays` with a format specified by `descriptor`.
```

Comment:
What should happen if a library does not support the specified format? This applies to libraries which support both dense and sparse arrays, as well as libraries such as NumPy which only support dense.

Reply:
Ah, right. This is an extension.

Reply:
Hmm, would you propose `xp.sparse.from_binsparse` instead? That way, the detection can also work via `hasattr(xp, "sparse")`.